### PR TITLE
Image Compression using PCA

### PR DESCRIPTION
**Pull request by DeewakarChakraborty**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
     
This fix is #3200     

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

File: content/articles/image-compression-using-PCA/index.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+> **Dimensionality reduction** is the transformation of data from a high-dimensional space into a low-dimensional space so that the low-dimensional representation retains some meaningful properties of the original data, ideally close to its intrinsic dimension(number of variables needed in a minimal representation of the data).
+
+Dimensionality reduction refers to techniques that reduce the number of input variables in a dataset. More input features often make a predictive modeling task more challenging to model, generally referred to as the <b>curse of dimensionality</b>.
+
+To explain the concept of dimensionality reduction, I will take an example, consider the following data where each point(vector) is represented using a linear combination of the **x** and **y** axes:
+
+<center>
+
+
+![first-graph](plot-1.png)
```
Comment (lalith1403) [inline content/articles/image-compression-using-PCA/index.md:63]: All images go with the following URL:

\[alt text\]\(/engineering-education/image-compression-using-PCA/image.type\)

File: content/articles/image-compression-using-PCA/index.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+</center>
+
+#### Motivation for Dimensionality Reduction
+
+Now, what if we choose a different basis
+
+<center>
+
+
+![image2](plot-2.jpg)
```
Comment (lalith1403) [inline content/articles/image-compression-using-PCA/index.md:74]: Same here as well

File: content/articles/image-compression-using-PCA/index.md [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+date: 2021-06-10T00:00:00-18:00
+topics: [Machine Learning]
+excerpt_separator: <!--more-->
+images:
+
+  -url: /engineering-education/image-compression-using-PCA/hero.png
+   alt: Image compression using PCA example image
+---
+
+Principal Component Analysis, PCA, is a dimensionality reduction method used to reduce the dimensionality of a dataset by transforming the data to a new basis where the dimensions are non-redundant(low covariance) and have high variance. This tutorial aims to make the reader understand the concept of PCA(Principal Component Analysis) mathematically by providing them with one of the use cases of PCA, image compression.
```
Comment (lalith1403) [inline content/articles/image-compression-using-PCA/index.md:18]: Explain the basic terminologies used in Linear Algebra:

1. row/column vectors
2. basis 
3. covariance
and other terms used.

File: content/articles/image-compression-using-PCA/index.md [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+Data correlation is how one set of data may correspond to another set. If two columns are highly correlated(or have high covariance), one of them is redundant since it is linearly dependent on the other column. We can normalize the correlation to get the correlation coefficient. The formula for the correlation coefficient is defined as :
+
+<center>
+
+
+![Equation](/engineering-education/image-compression-using-PCA/capture.jpg)
+
+</center>
+
+#### Desideratum
```
Comment (lalith1403) [inline content/articles/image-compression-using-PCA/index.md:97]: This heading may not be followed by everyone. Can you modify this?

File: content/articles/image-compression-using-PCA/index.md [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+- The dimensions are linearly <b>independent</b>(uncorrelated)
+- If we want to reduce dimensions by transforming the data into a new basis, that basis should be <b>orthogonal</b>.
+
+### Principal Component Analysis(PCA)
+
+> **Principal Component Analysis (PCA)** is a statistical procedure that uses an orthogonal transformation that converts a set of correlated variables to uncorrelated variables.
+
+To explain the concept of PCA mathematically, I will take an example:
+
+#### Orthogonal  Transformation
```
Comment (lalith1403) [inline content/articles/image-compression-using-PCA/index.md:111]: Remove additional space

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Hello @DeewakarChakraborty, Welcome to the Section EngEd Program. This is an amazing article. I thoroughly enjoyed reading through the article. It challenged me on various levels thus leading to greater learning objectives being met. Kudos to a fantastic first article. Look forward to many more! 

There are a few minor changes required on your end. Consider making them and ping me once done. 

Cheers,

Lalith

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/image-compression-using-PCA/index.md [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+
+Let p<sub>1</sub>, p<sub>2</sub>, p<sub>3</sub>,....,p<sub>n</sub> be a set of n orthonormal vectors. Let p be a square matrix of order n such that p<sub>1</sub>, p<sub>2</sub>, p<sub>3</sub>,....,p<sub>n</sub> are the columns of matrix p.
+
+Let x<sub>1</sub>, x<sub>2</sub>, x<sub>3</sub>,.....,x<sub>n</sub> &isin; R<sup>n</sup> be n data points and let X be a square matrix such that x<sub>1</sub>, x<sub>2</sub>, x<sub>3</sub>,.....,x<sub>n</sub>  are the rows of a matrix (Assumption: Data is zero-mean and unit variance), why we have this assumption we will come to this at later part.
+
+##### Transformation
+
+Suppose we want to represent x<sub>i</sub> using the new basis P.
+
+`x<sub>i</sub> = &prop;<sub>i1</sub>p<sub>1</sub> + &prop;<sub>i2</sub>p<sub>2</sub> + ....... + &prop;<sub>in</sub>p<sub>n</sub>`
```
Comment (lalith1403) [inline content/articles/image-compression-using-PCA/index.md:125]: This is how this equation is in preview. 
image

You will have to use in-line Latex for all the equations.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Modification is required in equations. Can you change them to Latex equations? Refer to this [PR.](https://github.com/section-engineering-education/engineering-education/pull/3101) Look at the inline equations used and replicate the same.

**Comment by hectorkambow** [comment]
@lalith1403  

Just wanted to get a status on this PR?

**Comment by hectorkambow** [comment]
Overall great work 👍 

Made some changes for format and flow
Please review to improve future articles and reviews. 

Please refrain from using HTML syntax - refer to any markdown for proper syntax 👍 

cc @lalith1403

**Pull request merged**